source/develop/release-management/features/virt/parallel-migration-connections.md

```diff
@@ -0,0 +1,88 @@
+
+- RDMA doesn't work.  This is not a limitation for oVirt, because RDMA is not supported in oVirt.
+
+- XBZRLE compression is not supported.  However this kind of compression is generally not recommended to use in migrations because it adds little benefit and a significant overhead.  gzip/zstd is supported instead and migrations should actually benefit from it.  (TODO: Check whether libvirt supports this kind of compression.)
+
+- Due to their better implementation, parallel migration connections can be faster than the traditional migration implementation even when the network bandwidth can be saturated with a single connection.  However, when single connection is specified, the parallel migration connection implementation uses a different code path, which may not be sufficiently robust, and it's recommended to always use at least two connections.  In the worst case, one of the threads will be idle while causing only a little overhead. 
+
+- Parallel migration connections haven't been tested extensively with connections faster than 100 Gbps.  Due to current multifd implementation, there is an increasing overhead when using many connections and it's not clear whether using more than 16 parallel connections adds a real benefit.
+
+- It still applies that large guests with terrabytes of RAM are difficult to migrate successfully.
+
+- Since it's easier to saturate network bandwidth with multiple connections, it's advisable to not migrate multiple VMs from a single host at the same time.
+
+- VM CPU pinning may restrict the number of CPUs available to the QEMU process and thus impose a limit on the meaningful number of parallel connections.
+
+- It's difficult to guess the right number of parallel connections to use.  A rule of thumb is to use one connection per 10 Gbps of available network bandwidth.  If the number of connections is a bit high, some threads may remain idle while the related overhead should be negligible.
+
+- There is very little documentation about parallel migration connections and the information above was obtained directly from platform developers.
+
+## User experience
+
+A new migration option to enable parallel migration connections is introduced in cluster and VM edit dialogs.  The cluster option specifies what to use in the given cluster.  VMs can override the cluster value individually.  The new option offers the following choices:
+
+- **Disabled** (the default in clusters): Parallel migration connections are not used.
+
+- **Auto**: Use parallel migration connections (TODO: Even when there is insufficient network bandwidth, for the supposed benefit of a more efficient implementation?) and set their number automatically based on the available resources (network bandwidth and CPU power).
+
+- **Custom**: Use parallel migration connections and use the number of connections specified by the user.  The minimum number of connection is 2.  (TODO: Should we put any real limit on the maximum number?)
```

Comment:
if it's debatable whether more than 16 connections adds a real benefit then we can add a configurable limit of 16 but anyway, even when the user sets a custom number it may make sense to reduce it based on the source/destination hosts, no?

Reply:
Do you mean the custom number of connections specified by the user would effectively set the maximum rather than a fixed number of connections?

Reply:
yeah, we have that heuristic of 10 Gbps per connection - I think that if the user specifies the number of connections, then we should not change it based on this heuristic but when it comes to CPU power, I think it never makes sense to use the specified number of connections when it exceeds the number of threads on the host(s), no?

Reply:
That's true. OK, I'll accommodate this observation here some way.

Reply:
yes, it blurs the difference between those options a bit but they will differ in case the number of vCPUs is greater than the amount of connections specified by the user or computed by the engine, and the amount of connections specified by the is different than the amount of connections that are computed by the engine based on the network bandwidth. it can go in both directions - that the user specifies a number that is greater than what the engine thinks is optimal (based on 10Gbps per connection) or lower than that

Reply:
Or to say it more simply to see the difference: If the user says "use 8 connections" then 8 connections will be used (unless it migrates between hosts that don't have that many threads, which is unlikely). With auto, it can be anything between 1 and the upper limit.